cpp/src/arrow/util/memory-pool-test.cc

```diff
@@ -46,4 +46,14 @@ TEST(DefaultMemoryPool, OOM) {
   ASSERT_RAISES(OutOfMemory, pool->Allocate(to_alloc, &data));
 }
 
+TEST(DefaultMemoryPoolDeathTest, FreeLargeMemory) {
+  MemoryPool* pool = default_memory_pool();
+
+  uint8_t* data;
+  ASSERT_OK(pool->Allocate(100, &data));
+  EXPECT_EXIT(pool->Free(data, 120), ::testing::ExitedWithCode(1),
+               ".*Check failed: \\(bytes_allocated_\\) >= \\(size\\)");
+  pool->Free(data, 100);
```

Comment:
Can you disable this test for release builds? See the `NDEBUG` define in `CMakeLists.txt`